pkg/snapstore/s3_snapstore.go

```diff
@@ -311,10 +355,25 @@ func (s *S3SnapStore) Save(snap brtypes.Snapshot, rc io.ReadCloser) error {
 	ctx, cancel := context.WithTimeout(ctx, chunkUploadTimeout)
 	defer cancel()
 	prefix := adaptPrefix(&snap, s.prefix)
-	uploadOutput, err := s.client.CreateMultipartUploadWithContext(ctx, &s3.CreateMultipartUploadInput{
-		Bucket: aws.String(s.bucket),
-		Key:    aws.String(path.Join(prefix, snap.SnapDir, snap.SnapName)),
-	})
+
+	var createMultipartUploadInput *s3.CreateMultipartUploadInput
+	if s.sseCustomerKey != "" {
+		// Customer managed Server Side Encryption
+		createMultipartUploadInput = &s3.CreateMultipartUploadInput{
+			Bucket:               aws.String(s.bucket),
+			Key:                  aws.String(path.Join(prefix, snap.SnapDir, snap.SnapName)),
+			SSECustomerAlgorithm: aws.String(s.sseCustomerAlgorithm),
+			SSECustomerKey:       aws.String(s.sseCustomerKey),
+			SSECustomerKeyMD5:    aws.String(s.sseCustomerKeyMD5),
+		}
+	} else {
+		// AWS managed Server Side Encryption
+		createMultipartUploadInput = &s3.CreateMultipartUploadInput{
+			Bucket: aws.String(s.bucket),
+			Key:    aws.String(path.Join(prefix, snap.SnapDir, snap.SnapName)),
+		}
+	}
```

Comment:
@seshachalam-yv suggests that this isn't the most elegant way to add fields to this struct, since there is repetition. The additional fields for SSE-C can just be added if SSE-C is enabled in its own `if`.